### PR TITLE
Implement CI/CD via Github Actions and Helm Charts

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR implements our typical Helm Chart and Github Actions based CI/CD pipeline. Environment values and other general configurations have been standardized to align with application functionality and value management lifecycles.

- 5914083 Implement remaining Github Actions deployment automation
- 1de99a4 Update dockerfile to properly support knex and prisma lifecycles
- 4e43e81 Implement PCNS Helm Chart with Patroni dependency

[SHOWCASE-3446](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-3446)

## Types of changes

New feature (non-breaking change which adds functionality)
Documentation (non-breaking change with enhancements to documentation)
Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by kyle1morel — APPROVED** [review]

File: .github/environments/values.prod.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    FRONTEND_COMS_APIPATH: https://coms.api.gov.bc.ca/api/v1
+    FRONTEND_OIDC_AUTHORITY: https://loginproxy.gov.bc.ca/auth/realms/standard
+    FRONTEND_OIDC_CLIENTID: nr-permit-connect-navigator-service-5188
+    SERVER_APIPATH: /api/v1
+    SERVER_BODYLIMIT: 30mb
+    SERVER_CHEFS_APIPATH: https://submit.digital.gov.bc.ca/app/api/v1
+    SERVER_DB_PORT: "5432"
+    # SERVER_LOGFILE: ~
+    SERVER_LOGLEVEL: http
+    SERVER_OIDC_AUTHORITY: https://dev.loginproxy.gov.bc.ca/auth/realms/standard
```
Comment (norrisng-bc) [inline .github/environments/values.prod.yaml:15]: Isn't this the dev Keycloak server?

File: .github/environments/values.test.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    FRONTEND_COMS_APIPATH: https://coms-test.api.gov.bc.ca/api/v1
+    FRONTEND_OIDC_AUTHORITY: https://test.loginproxy.gov.bc.ca/auth/realms/standard
+    FRONTEND_OIDC_CLIENTID: nr-permit-connect-navigator-service-5188
+    SERVER_APIPATH: /api/v1
+    SERVER_BODYLIMIT: 30mb
+    SERVER_CHEFS_APIPATH: https://submit.digital.gov.bc.ca/app/api/v1
+    SERVER_DB_PORT: "5432"
+    # SERVER_LOGFILE: ~
+    SERVER_LOGLEVEL: http
+    SERVER_OIDC_AUTHORITY: https://dev.loginproxy.gov.bc.ca/auth/realms/standard
```
Comment (norrisng-bc) [inline .github/environments/values.test.yaml:15]: Isn't this the dev Keycloak server?

File: .github/environments/values.prod.yaml (hunk 1 above)
Comment (jujaga) [reply .github/environments/values.prod.yaml:15]: Lol whoops good catch 😅

File: .github/environments/values.test.yaml (hunk 2 above)
Comment (jujaga) [reply .github/environments/values.test.yaml:15]: Lol whoops good catch 😅

**Review by norrisng-bc — APPROVED** [review]

**Pull request merged**